### PR TITLE
Feature/160056372/signup

### PR DESCRIPTION
**Pull request by Fattylee**
#### What does this PR do?
Implement local signup

#### Description of Task to be completed?
- create signup form component
- write signup logic
- connect signup page to redux
- handle error and success
- write tests for components

#### How should this be manually tested?
- run `npm install`
- run `npm test`

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
#160056372

#### Screenshots (if appropriate)

screen shot 2018-09-14 at 1 14 06 pm

screen shot 2018-09-14 at 5 40 04 pm

screen shot 2018-09-14 at 5 43 50 pm

screen shot 2018-09-14 at 5 43 50 pm

#### Questions:
N/A

### CODE REVIEW

**Review by chidioguejiofor — APPROVED** [review]

File: src/actions/authAction.js [hunk 1]
```diff
@@ -62,10 +62,17 @@ export const validateResetToken = path => ({
 });
 
 
+const signup = formData => ({
+  type: types.SIGN_UP,
+  payload: axiosInstance()
+    .post('http://localhost:3000/api/users/', { user: formData }),
```
Comment (chidioguejiofor) [inline src/actions/authAction.js]: Use axios instance here

File: src/components/SocialLogin.jsx [hunk 2]
```diff
@@ -3,9 +3,9 @@ import PropTypes from 'prop-types';
 
 const SocialLogin = ({ socialMedia }) => {
   let host = 'https://authorhaven.herokuapp.com';
-  if (process.env.NODE_ENV === 'development') {
-    host = 'http://localhost:3000';
-  }
+  // if (process.env.NODE_ENV === 'development') {
+  //   host = 'http://localhost:3000';
+  // }
```
Comment (chidioguejiofor) [inline src/components/SocialLogin.jsx]: remove these comments

**Review by mppdg — APPROVED** [review]

File: src/middleware/authMiddleware.js [hunk 3]
```diff
@@ -24,10 +32,10 @@ const authMiddleware = () => next => (action) => {
     localStorage.removeItem('token');
     localStorage.removeItem('user');
   }
-  if (action.type === types.SOCIAL_LOGIN) {
-    const { token } = action.payload.user;
-    localStorage.setItem('token', token);
-  }
+  // if (action.type === types.SOCIAL_LOGIN) {
```
Comment (rhenshaw56) [inline src/middleware/authMiddleware.js]: remove the commented bit out

File: src/actions/authAction.js [hunk 4]
```diff
@@ -62,10 +62,17 @@ export const validateResetToken = path => ({
 });
 
 
+const signup = formData => ({
+  type: types.SIGN_UP,
+  payload: axiosInstance()
```
Comment (rhenshaw56) [inline src/actions/authAction.js]: you can easily do `axiosInstance.post`, not sure why you're invoking axiosInstance

**Review by rhenshaw56 — APPROVED** [review]
Just fix the two comments I raised and this should be good to go

File: src/actions/authAction.js (hunk 4 above)
Comment (Fattylee) [reply src/actions/authAction.js]: The axioInstance is a function, not an object. 
` payload: axiosInstance().post('/users/', { user: formData }),`

File: src/middleware/authMiddleware.js (hunk 3 above)
Comment (Fattylee) [reply src/middleware/authMiddleware.js]: Okay. removed.

**Pull request merged**